src/connectivity.js

```diff
@@ -137,17 +155,42 @@ var ConnectivityChecker = {
         return reqOptions;
       };
 
+
       if (RdGlobalConfig.proxy) {
+        var proxyOpts = url.parse(RdGlobalConfig.proxy.host + ":" +RdGlobalConfig.proxy.port);
+        if (RdGlobalConfig.proxy.username && RdGlobalConfig.proxy.password) {
+          proxyOpts.auth = RdGlobalConfig.proxy.username + ":" + RdGlobalConfig.proxy.password;
+        }
+
         ConnectivityChecker.connectionChecks.push(this.httpToHubWithProxy, this.httpToRailsWithProxy);
         /* eslint-disable-next-line no-unused-vars */
         ConnectivityChecker.reqOpsWithProxy = function (reqUrl, reqType) {
```

Comment:
we can rename the function to `reqOpsHttpWithProxy` since we have the `https` function separate now...

Reply:
Done

Reply:
can be resolved.